scripts/ci/verify-manifest-tools.py

```
@@ -83,13 +99,14 @@ def _tool_command(tool_name: str, install: dict[str, Any]) -> list[str]:
     return [bin_name or tool_name, "--version"]
 
 
-def _load_manifest(path: str) -> list[dict[str, Any]]:
+def _load_manifest(path: str) -> tuple[list[dict[str, Any]], int]:
     with open(path, encoding="utf-8") as handle:
         data = json.load(handle)
     tools = data.get("tools", [])
     if not isinstance(tools, list):
         raise ValueError("manifest tools must be a list")
```

Ruff (0.15.7)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)